scripts/hera-sim-vis.py

```diff
@@ -0,0 +1,164 @@
+#!/bin/env python
+
+"""
+Command-line interface for simulating visibilities with ``hera_sim``.
+
+This script may be used to run a visibility simulation from a configuration file and
+write the result to disk.
+"""
+import sys
+import yaml
+import argparse
+from pathlib import Path
+
+import numpy as np
+import pyuvdata
+import pyuvsim
+import pyradiosky
+import hera_sim
+from mpi4py import MPI
+from hera_sim.visibilities import (
+    ModelData,
+    VisibilitySimulation,
+    load_simulator_from_yaml,
+)
+from rich.console import Console
+from rich.rule import Rule
+from rich.panel import Panel
+
+cns = Console()
+
+
+def cprint(*args, **kwargs):
+    """Print only if root worker."""
+    if myid == 0:
+        cns.print(*args, **kwargs)
+
+
+parser = argparse.ArgumentParser(
+    description="Run vis_cpu via hera_sim given an obsparam."
+)
+parser.add_argument("obsparam", type=str, help="pyuvsim-formatted obsparam file.")
+parser.add_argument(
+    "simulator_config", type=str, help=" YAML configuration file for the simulator."
+)
+parser.add_argument(
+    "--object_name", type=str, default=None, help="Set object_name in the UVData"
+)
+parser.add_argument("--compress", action="store_true", help="Compress by redundancy.")
+parser.add_argument(
+    "--normalize_beams", action="store_true", help="Peak normalize the beams."
+)
+parser.add_argument(
+    "--fix_autos", action="store_true", help="Check and fix non-real xx/yy autos"
+)
+parser.add_argument(
+    "--max-auto-imag",
+    type=float,
+    default=5e-14,
+    help="Maximum fraction of imaginary/absolute for autos before raising an error.",
+)
+
```

Comment:
Do you have thoughts on whether scripts should generally have their contents protected by a `if __name__ == "__main__":` clause?

Reply:
yeah, it probably should. Done.